src/app/core/i18n/i18n.service.ts

```diff
@@ -5,85 +5,95 @@ import {loadTranslations} from "@angular/localize";
 @Injectable({
     providedIn: "root",
   })
-  export class I18n {
-    locale = "en";
-    readonly allowedLocales = ["en", "fr", "de", "it"];
-    translations: any = {};
-  
-    async setLocale() {
-      const userLocale = localStorage.getItem("locale");
-  
-      // If the user has a preferred language stored in LocalStorage, use it.
-      if (userLocale && this.allowedLocales.includes(userLocale)) {
-        this.locale = userLocale;
-      } else {
-        localStorage.setItem("locale", this.locale);
-      }
-  
-      // Use webpack magic string to only include required locale data
-      const localeModule = await import(
-        /* webpackInclude: /(en|de|fr|it)\.mjs$/ */
-        `/node_modules/@angular/common/locales/${this.locale}.mjs`
-      );
-      registerLocaleData(localeModule.default);
-  
-      // Load translation file initially
-      await this.loadTranslations();
-    }
-  
-    async loadTranslations() {
-      const localeTranslationsModule = await import(
-        `src/assets/i18n/${this.locale}.json`
-      );
-  
-      // Ensure translations are flattened if necessary
-      this.translations = this.flattenTranslations(localeTranslationsModule.default);
-  
-      // Load translations for the current locale at runtime
-      loadTranslations(this.translations);
+export class I18nService {
+  readonly allowedLanguages = ["en", "fr", "de", "it"];
+  private current_language: string = this.getLanguageFromStorage() || this.detectNavigatorLanguage();
```

Comment:
Nit: camelCase is used in NGE instead of snake_case

Reply:
oups